uat/testing-features/src/main/java/com/aws/greengrass/steps/CloudWatchSteps.java

```diff
@@ -80,34 +100,107 @@ public void deleteLogGroup(String componentType, String componentName) {
+     * @param timeout       Number of seconds to wait before timing out the operation
+     * @throws InterruptedException {@link InterruptedException}
      */
 
-    @Then("I verify that it created a log group for component type {word} for component {word}, with streams within "
+    @Then("I verify that it created a log group of type type {word} for component {word}, with streams within "
             + "{int} seconds in CloudWatch")
     public void verifyCloudWatchGroupWithStreams(String componentType, String componentName, int timeout) throws
-            InterruptedException {
+            Exception {
         SimpleDateFormat formatter = new SimpleDateFormat("yyyy/MM/dd", Locale.ENGLISH);
         formatter.setTimeZone(TimeZone.getTimeZone("UTC")); // All dates are UTC, not local time
 
-        String thingName = testContext.coreThingName();
-        String region = resourceContext.region().toString();
-
-        String logGroupName = String.format("/aws/greengrass/%s/%s/%s", componentType, region, componentName);
-        String logStreamNamePattern = String.format("/%s/thing/%s", formatter.format(new Date()), thingName);
+        String logGroupName = getLogGroupName(componentType, componentName);
+        String logStreamName = getLogStreamName();
 
         resources.create(CloudWatchLogStreamSpec
                 .builder()
                 .logGroupName(logGroupName)
-                .logStreamName(logStreamNamePattern)
+                .logStreamName(logStreamName)
                 .build());
 
-        LOGGER.info("Verifying log group {} with stream {} was created", logGroupName, logStreamNamePattern);
-        waitSteps.untilTrue(() -> doesStreamExistInGroup(logGroupName, logStreamNamePattern), timeout,
+        LOGGER.info("Verifying log group {} with stream {} was created", logGroupName, logStreamName);
+        boolean result = waitSteps.untilTrue(() -> doesStreamExistInGroup(logGroupName, logStreamName), timeout,
                 TimeUnit.SECONDS);
+
+        if (!result) {
+            throw new Exception(String.format("Failed to find log stream %s within group %s",
+                    logStreamName, logGroupName));
+        }
+    }
+
+
+    @Then("I verify {int} logs for {word} of type {word} have been uploaded to Cloudwatch within {int} seconds")
+    public void verifyLogs(int numberOfLogLines, String componentName, String componentType, int timeout) throws
+            Exception {
+        boolean result = waitSteps.untilTrue(() -> haveAllLogsBeenUploaded(numberOfLogLines, componentName,
+                componentType), timeout, TimeUnit.SECONDS);
+
+        if (!result) {
+            // Print all the cloudwatch logs it fetched, so we can debug what failed to get uploaded.
+            this.lastReceivedCloudWatchEvents.forEach(e -> LOGGER.info(e.message()));
+
+            throw new Exception(String.format("Failed to verify that %s logs were uploaded to CloudWatch",
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: numberOfLogLines (declared at line 137). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 146. Consider formatting this data with &#39;%d&#39; instead.